File: TestApp/app/src/main/java/com/tsi84/testapp/dialog/LoadingDialog.java
package com.tsi84.testapp.dialog;

import android.app.Dialog;
import android.content.Context;
import android.view.Window;
import android.widget.TextView;

import com.tsi84.testapp.R;

public class LoadingDialog extends Dialog {

	private TextView mTextMessage;

	public LoadingDialog(Context context) {
		this(context, null);
	}

	public LoadingDialog(Context context, String message) {
		super(context);

		// set property
		requestWindowFeature(Window.FEATURE_NO_TITLE);

		// layout
		setContentView(R.layout.loading_dialog);
		mTextMessage = (TextView) findViewById(R.id.progress_text);

		if (message != null) {
			mTextMessage.setText(message);
		}
	}

	public void setTextMessage(String message) {
		if (mTextMessage != null) {
			mTextMessage.setText(message);
		}
	}
}

File: mentoring/TestKotlin/app/src/main/java/com/tsi84/testkotlin/dialog/LoadingDialog.kt
package com.tsi84.testkotlin.dialog

import android.app.Dialog
import android.content.Context
import android.view.View
import android.view.Window
import android.widget.TextView
import com.tsi84.testkotlin.R

class LoadingDialog @JvmOverloads constructor(context: Context?, message: String? = null) : Dialog(context!!) {
    private val mTextMessage: TextView?
    fun setTextMessage(message: String?) {
        if (mTextMessage != null) {
            mTextMessage.text = message
        }
    }

    init {

        // set property
        requestWindowFeature(Window.FEATURE_NO_TITLE)

        // layout
        setContentView(R.layout.loading_dialog)
        mTextMessage = findViewById<View>(R.id.progress_text) as TextView
        message?.let { setTextMessage(it) }
    }
}